ttax/decompositions.py

```diff
@@ -0,0 +1,181 @@
+    raise ValueError('max_tt_rank should be a number or a vector of size (d+1) '
+                     'where d is the number of dimensions (rank) of the tensor.')
+  if tt.is_tt_matrix:
+    raw_shape = tt.raw_tensor_shape
+  else:
+    raw_shape = tt.shape
+
+  tt_cores = orthogonalize_tt_cores(tt).tt_cores
+  # Copy cores references so we can change the cores.
+  tt_cores = list(tt_cores)
+
+  ranks = [1] * (num_dims + 1)
+  are_tt_ranks_defined = True
+
+  # Right to left SVD compression.
+  for core_idx in range(num_dims - 1, 0, -1):
+    curr_core = tt_cores[core_idx]
+    if tt.is_tt_matrix:
+      curr_mode_left = raw_shape[0][core_idx]
+      curr_mode_right = raw_shape[1][core_idx]
+      curr_mode = curr_mode_left * curr_mode_right
+    else:
+      curr_mode = raw_shape[core_idx]
+    
+    columns = curr_mode * ranks[core_idx + 1]
+    curr_core = jnp.reshape(curr_core, [-1, columns])
+    #rows = curr_core.shape.as_list()[0]
+    rows = list(curr_core.shape)[0]
+    if rows is None:
+      rows = jnp.shape(curr_core)[0]
+    if max_tt_rank[core_idx] == 1:
+      ranks[core_idx] = 1
+    else:
+      try:
+        ranks[core_idx] = min(max_tt_rank[core_idx], rows, columns)
+      except TypeError:
+        # Some of the values are undefined on the compilation stage and thus
+        # they are tensors (?) instead of values.
+        min_dim = jnp.minimum(rows, columns)
+        ranks[core_idx] = jnp.minimum(max_tt_rank[core_idx], min_dim)
+        are_tt_ranks_defined = False
+    u, s, v = jnp.linalg.svd(curr_core, full_matrices=False)
+    u = u[:, 0:ranks[core_idx]]
+    s = s[0:ranks[core_idx]]
+    v = v[:, 0:ranks[core_idx]]
+    if tt.is_tt_matrix:
+      core_shape = (ranks[core_idx], curr_mode_left, curr_mode_right,
+                    ranks[core_idx + 1])
+    else:
+      core_shape = (ranks[core_idx], curr_mode, ranks[core_idx + 1])
+    tt_cores[core_idx] = jnp.reshape(jnp.transpose(v), core_shape)
+    prev_core_shape = (-1, rows)
+    tt_cores[core_idx - 1] = jnp.reshape(tt_cores[core_idx - 1], prev_core_shape)
+    tt_cores[core_idx - 1] = jnp.matmul(tt_cores[core_idx - 1], u)
+    tt_cores[core_idx - 1] = jnp.matmul(tt_cores[core_idx - 1], jnp.diag(s))
+
+  if tt.is_tt_matrix:
+    core_shape = (ranks[0], raw_shape[0][0], raw_shape[1][0], ranks[1])
+  else:
+    core_shape = (ranks[0], raw_shape[0], ranks[1])
+  tt_cores[0] = jnp.reshape(tt_cores[0], core_shape)
+  if not are_tt_ranks_defined:
+    ranks = None
+  return TT(tt_cores)
+
+
+  def orthogonalize_tt_cores(tt, left_to_right=True,
+                           name='ttax_orthogonalize_tt_cores'):
+  """Orthogonalize TT-cores of a TT-object.
+  Args:
+    tt: TenosorTrain or a TensorTrainBatch.
+    left_to_right: bool, the direction of orthogonalization.
+    name: string, name of the Op.
+  Returns:
+    The same type as the input `tt` (TenosorTrain or a TensorTrainBatch).
+  """
+  # Left to right orthogonalization.
```

Comment:
add something like
if not left_to_right:
  raise NotImplementedError('Right to left orthogonalization is not supported')